src/metaxy/versioning/engine.py

```diff
@@ -461,9 +493,20 @@ def hash_struct_version_column(
         field_names = sorted([f.key.to_struct_key() for f in self.plan.feature.fields])
 
         # Concatenate all field hashes with separator
-        sample_components = [
-            nw.col(struct_column).struct.field(field_name) for field_name in field_names
-        ]
+        sample_components = []
+        for field_name in field_names:
+            expr = self.access_provenance_field(struct_column, field_name)
+            # If data_version is missing, fall back to provenance-by-field (useful for dict-based stores)
```

Comment:
I'm not sure we are supposed to be handling this, if it's missing something went really wrong. 

write_metadata always ensures this column is present so it can't happen. 

In tests it's our responsibility to provide it. We have a hypothesis strategy that does it.